pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -27,9 +27,93 @@ import (
 )
 
 func (e *AzureLogAnalyticsDatasource) ResourceRequest(rw http.ResponseWriter, req *http.Request, cli *http.Client) (http.ResponseWriter, error) {
+	if req.URL.Path == "/usage/basiclogs" {
+		newUrl := &url.URL{
+			Scheme: req.URL.Scheme,
+			Host:   req.URL.Host,
+			Path:   "/v1/query",
+		}
+		return e.GetBasicLogsUsage(req.Context(), newUrl.String(), cli, rw, req.Body)
+	}
 	return e.Proxy.Do(rw, req, cli)
 }
 
+// builds and executes a new query request that will get the data ingeted for the given table in the basic logs query
+func (e *AzureLogAnalyticsDatasource) GetBasicLogsUsage(ctx context.Context, url string, client *http.Client, rw http.ResponseWriter, reqBody io.ReadCloser) (http.ResponseWriter, error) {
+	// read the full body
+	originalPayload, readErr := io.ReadAll(reqBody)
+	if readErr != nil {
+		return rw, fmt.Errorf("failed to read request body %w", readErr)
+	}
+	var payload BasicLogsUsagePayload
+	jsonErr := json.Unmarshal(originalPayload, &payload)
+	if jsonErr != nil {
+		return rw, fmt.Errorf("when getting basic logs table usage, failed to decode the query object from JSON: %w", jsonErr)
+	}
+	table := payload.Table
+
+	from, fromErr := ConvertTime(payload.From)
+	if fromErr != nil {
+		return rw, fmt.Errorf("failed to convert from time: %w", fromErr)
+	}
+
+	to, toErr := ConvertTime(payload.To)
+	if toErr != nil {
+		return rw, fmt.Errorf("failed to convert to time: %w", toErr)
+	}
+
+	dataVolumeQueryRaw := GetDataVolumeRawQuery(table)
+	dataVolumeQuery := &AzureLogAnalyticsQuery{
+		Query:         dataVolumeQueryRaw,
+		DashboardTime: true,
+		TimeRange: backend.TimeRange{
+			From: from,
+			To:   to,
+		},
+		TimeColumn: "TimeGenerated",
+		Resources:  []string{payload.Resource},
+		QueryType:  dataquery.AzureQueryType(payload.QueryType),
+		URL:        getApiURL(payload.Resource, false),
+	}
+
+	req, err := e.createRequest(ctx, url, dataVolumeQuery)
+	if err != nil {
+		return rw, err
+	}
+	resp, err := client.Do(req)
```

Comment:
Are you able to add tracing to this request please? Example [here](https://github.com/grafana/grafana/blob/bfc6abe9c405cc506b19bf78634c5ef2a8372fa2/pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go#L162)